internal/pkg/csumio/csumwriter.go

```diff
@@ -0,0 +1,92 @@
+package csumio
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+const (
+	ChecksumLen      = 8
+	minimumChunkSize = 4 * 1024 // 4 KiB
+)
+
+type ChecksumWriter struct {
```

Comment:
Since it's in the package `csumio` and it's obvious that the writer is related to checksum, how about renaming `ChecksumWriter` to `Writer` ?